xcc/settings.py

```diff
@@ -3,12 +3,19 @@
 """
 
 import os
+import re
 from typing import Optional
 
 from appdirs import user_config_dir
 from dotenv import dotenv_values, set_key, unset_key
 from pydantic import BaseSettings
 
+# Matches when string contains chars outside Base64URL set
+# https://base64.guru/standards/base64url
+# https://en.wikipedia.org/wiki/Base64#Variants_summary_table
+# Also, the dot '.' to separate sections.
+_BASE64URLRE = re.compile(r"[^\.A-Za-z0-9_-]+")
```

Comment:
What is the motivation behind making this a global variable?

Reply:
That's the compiled regex that we're looking for (and hoping _not_ to find, because it represents characters _not_ in the acceptable set). I suppose it doesn't have to be compiled, we can just do it inline. _I felt that compiling it and giving it a name helped document what it was, rather than expressing it inline in the function_. It doesn't really matter.